File: AngularAndNetCoreAuth/Models/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace AngularAndNetCoreAuth.Models
{
    public class Student :BaseEntity
    {
        [Key]
        public int ID { set; get; }
        // refers to ID in srms student table
        [Required]
        public string StudentId { set; get; }
        [Required]
        public string Stno { set; get; }
        [Required]
        public string IndexNo { set; get; }
        [Required]
        public string Name { set; get; }
        [Required]
        public string Programme { set; get; }
        [Required]
        public string Level { set; get; }
        [Required]
        public string Phone { set; get; }
        [Required]
        public string Email { set; get; }
      
        public virtual UserData user { get; set; }

    }
}

File: New_Service/backend/src/main/java/com/ttu/pay/model/Student.java
package com.ttu.pay.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;
import java.util.UUID;
@Getter
@Setter
@Accessors(chain=true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Student{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;
    @NotNull
    public String studentId;
    @NotNull
    public String stno;
    @NotNull
    public String indexNo;
    @NotNull
    public String name ;

    @NotNull
    public String programme;
    @NotNull
    public String level;
    @NotNull
    public String phone;
    @NotNull
    public String email ;
    @OneToOne
    @JoinColumn(name = "user_id")
    @NotNull
    public User user;
    private LocalDateTime CreatedOn;
    private LocalDateTime UpdatedOn;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}
